parameters/examples/genesis.rs

```diff
@@ -58,71 +58,51 @@ pub fn generate<C: Parameters>(recipient: &Address<C>, value: u64) -> Result<(Ve
     let private_keys = vec![genesis_account.private_key.clone(); C::NUM_INPUT_RECORDS];
 
     let mut joint_serial_numbers = Vec::with_capacity(C::NUM_INPUT_RECORDS);
-    let mut old_records = Vec::with_capacity(C::NUM_INPUT_RECORDS);
+    let mut input_records = Vec::with_capacity(C::NUM_INPUT_RECORDS);
     for i in 0..C::NUM_INPUT_RECORDS {
-        let old_record = Record::new(
-            &dpc.noop_program,
-            genesis_account.address.clone(),
-            true, // The input record is a noop.
-            0,
-            Payload::default(),
-            C::serial_number_nonce_crh().hash(&[64u8 + (i as u8); 1])?,
-            rng,
-        )?;
-
-        let (sn, _) = old_record.to_serial_number(&private_keys[i])?;
+        let input_record = Record::new_noop_input(&dpc.noop_program, genesis_account.address, rng)?;
+
+        let (sn, _) = input_record.to_serial_number(&private_keys[i])?;
         joint_serial_numbers.extend_from_slice(&to_bytes_le![sn]?);
 
-        old_records.push(old_record);
+        input_records.push(input_record);
     }
 
-    // Construct the new records.
-    let mut new_records = Vec::with_capacity(C::NUM_OUTPUT_RECORDS);
-    new_records.push(Record::new_full(
+    // Construct the output records.
+    let mut output_records = Vec::with_capacity(C::NUM_OUTPUT_RECORDS);
+    output_records.push(Record::new_output(
         &dpc.noop_program,
-        recipient.clone(),
+        recipient,
         false,
         value,
         Payload::default(),
         C::NUM_INPUT_RECORDS as u8,
         joint_serial_numbers.clone(),
         rng,
     )?);
-    new_records.push(Record::new_full(
```

Comment:
I agree the input/output records are better than old/new records, as the old/new records may give people a feeling that we are just updating existing records.